SPECS-EXTENDED/reaper/reaper.spec

```diff
@@ -0,0 +1,156 @@
+Summary: Cassandra Reaper
+Name: reaper
+Version: 3.1.1
+Release: 1%{?dist}
+License: Apache 2.0
+Vendor: Microsoft Corporation
+Distribution: Mariner
+Group: Applications/System
+
+URL: http://cassandra-reaper.io/
+Source0: https://github.com/thelastpickle/cassandra-reaper/archive/refs/tags/%{version}.tar.gz
+Source1: cassandra-reaper-%{version}.tar.gz
+# Building reaper from sources downloads artifacts related to maven/node/etc. These artifacts need to be downloaded as caches in order to build reaper using maven in offline mode.
+# Below is the list of cached sources.
+# bower-components downloaded under src/ui
+Source2: reaper-bower-components-%{version}.tar.gz
+# node_modules downloaded under src/ui
+Source3: reaper-srcui-node-modules-%{version}.tar.gz
+# bower cache
+Source4: reaper-bower-cache-%{version}.tar.gz
+# m2 cache
+Source5: reaper-m2-cache-%{version}.tar.gz
+# npm cache
+Source6: reaper-npm-cache-%{version}.tar.gz
+# node_modules downloaded to /usr/local/lib
+Source7: reaper-local-lib-node-modules-%{version}.tar.gz
+# v14.18.0 node binary under /usr/local
+Source8: reaper-local-n-%{version}.tar.gz
```

Comment:
For cache tarballs please provide steps to reproduce them. Ideally in form of a script(s) automating the whole process, so we know where these came from and how to create a fresh batch after a version update.

[Preferred] Example of script usage: `SPECS/azure-iotedge/azure-iotedge.spec` (uses `toolkit/scripts/build_cargo_cache.sh`).

[Discouraged] Example of in-spec comments with repro steps: `SPECS/cert-manager/cert-manager.spec`.

Reply:
These sources are artifacts that are downloaded when building in online mode. In my opinion, should I move the steps to generate these tar balls inside the spec file itself? There is a separate "**Readme**" file listing down all the steps.

Reply:
The packages build without network access, so it's better to just provide a comment with steps or a script one can run outside of the regular package build. In one spec I introduced a separate path for generating the cache through a package build when a specific macro was set but I regret that decision now - it's adding a lot of noise to the spec.

If you're saying there's a readme with the required steps, you can mention that in a comment here but a script one could run with `build_cache.sh [version]` and get all of that done automatically would be nicer. The script could live next to this spec, if it's specific only to this package.

Reply:
The cache generation is related to this spec only. However, in order to generate cache tars, one needs to build reaper sources itself. I have added a 'generate_build_caches' flag in the spec file itself, which creates the build cache tars and packs them into a reaper-build-cache rpm. Only catch is this approach will work in local builds with RUN_CHECK=y, thus enabling dns resolution to be able to download the artifacts.

Reply:
Adding these additional bits guarded by `generate_build_caches` is what I'd like to avoid in the spec - it's additional noise not related to building the package itself, closer to a hack - we don't really want the `build-caches` subpackage, we won't be publishing it to our RPMs repository, and it's somewhat cumbersome to extract its contents to get the sources we need. In addition to that using this approach requires us to modify the spec and build the whole package locally. Alternatively, our pipeline test builds would end up building additional packages, which may cause some confusion down the road.

Since you already have the steps for building the cache tarballs inside the spec, is there anything stopping us from running these steps separately from a script whenever we need update the version? It would be also faster than doing a whole package build.

Reply:
Created "reaper_build_caches.sh" to aide in re-generating build caches.
Cleaned conditional checks from SPEC file